File: 3626-smallest-divisible-digit-product-i/3626-smallest-divisible-digit-product-i.cpp
class Solution {
public:
    int smallestNumber(int n, int t) {
        while (true) {
            int k = 1;
            int i = n;
            while (i) {
                int rem = i % 10;
                k *= rem;
                i /= 10;
                if (k % t == 0) {
                    return n;
                }
                
            }
            n++;
        }
        return n;
    }
};

File: 3626-smallest-divisible-digit-product-i/3626-smallest-divisible-digit-product-i.py
class Solution:
    def smallestNumber(self, n: int, t: int) -> int:
        while(True):
            k=1
            i=n
            while(i):
                rem=i%10
                k*=rem
                i//=10
            if(k%t==0):
                return n
            n+=1
        